python/cugraph/cugraph/gnn/graph_store.py

```diff
@@ -210,67 +211,99 @@ def sample_neighbors(
 
-        del dst_seeds, offset_lens, seeds_offsets_s
-        del ego_edge_list, seeds_offsets
+        if edge_dir == "in":
+            sg = self.extracted_reverse_subgraph_without_renumbering
+        else:
+            sg = self.extracted_subgraph_without_renumbering
 
-        # Step 2
-        # Sample Fan Out
-        # for each dst take maximum of fanout samples
-        filtered_list = sample_groups(
-            filtered_list, by="dst", n_samples=fanout
+        sampled_df = uniform_neighbor_sample(
+            sg, start_list=nodes, fanout_vals=[fanout],
+            with_replacement=replace
         )
 
-        # TODO: Verify order of execution
-        sample_df = cudf.DataFrame(
-            {src_n: filtered_list["src"], dst_n: filtered_list["dst"]}
-        )
-        del filtered_list
+        sampled_df.drop(columns=["indices"], inplace=True)
 
-        # del parents_nodes, children_nodes
-        edge_df = sample_df.merge(
-            self.gdata._edge_prop_dataframe[[src_n, dst_n, eid_n]],
-            on=[src_n, dst_n],
-        )
+        # handle empty graph case
+        if len(sampled_df) == 0:
+            return None, None, None
+
+        # we reverse directions when directions=='in'
+        if edge_dir == "in":
+            sampled_df.rename(
+                columns={"destinations": src_n, "sources": dst_n}, inplace=True
+            )
+        else:
+            sampled_df.rename(
+                columns={"sources": src_n, "destinations": dst_n}, inplace=True
+            )
+
+        edge_df = self.gdata._edge_prop_dataframe[[src_n, dst_n, eid_n]]
+        sampled_df = edge_df.merge(sampled_df)
 
         return (
-            edge_df[src_n].to_dlpack(),
-            edge_df[dst_n].to_dlpack(),
-            edge_df[eid_n].to_dlpack(),
+            sampled_df[src_n].to_dlpack(),
+            sampled_df[dst_n].to_dlpack(),
+            sampled_df[eid_n].to_dlpack(),
         )
 
-    def find_edges(self, edge_ids, etype):
+    @cached_property
+    def extracted_reverse_subgraph_without_renumbering(self):
+        subset_df = self.gdata._edge_prop_dataframe[[src_n, dst_n]]
+        subset_df.rename(columns={src_n: dst_n, dst_n: src_n}, inplace=True)
+        subset_df["weight"] = cp.float32(1.0)
+        subgraph = cugraph.Graph(directed=True)
+        subgraph.from_cudf_edgelist(
+            subset_df,
+            source=src_n,
+            destination=dst_n,
+            edge_attr="weight",
+            renumber=False,
+        )
+        return subgraph
+
+    @cached_property
+    def extracted_subgraph_without_renumbering(self):
+        subset_df = self.gdata._edge_prop_dataframe[[src_n, dst_n]]
+        subset_df["weight"] = cp.float32(1.0)
+        subgraph = cugraph.Graph(directed=True)
+        subgraph.from_cudf_edgelist(
+            subset_df,
+            source=src_n,
+            destination=dst_n,
+            edge_attr="weight",
+            renumber=False,
```

Comment:
May need to switch to `renumber=True`.

See issue: https://github.com/rapidsai/cugraph/issues/2446

Reply:
renumber=True. still is causing errors. Todo: Raise a MRE without it and comment to the issue.

Reply:
legacy_renum_only=True in graph creation